Parameterize docker registry URL

irregulator opened the pull request:
This will allow Jenkins to swap the variable accordingly.

Commit:
Parameterize docker registry URL

This will allow Jenkins to swap the variable accordingly.

irregulator commented:
[sc-61884]

xgattuso commented on Jenkinsfile:
```diff
@@ -12,7 +12,7 @@ node {
 
       stage('Build debian package') {
-	sh 'make jammy_package'
+	sh 'make jammy_package EXOSCALE_DOCKER_REGISTRY=${EXOSCALE_DOCKER_REGISTRY}'
```
```suggestion
	sh 'make jammy_package EXOSCALE_DOCKER_REGISTRY=${EXOSCALE_DOCKER_REGISTRY}'
```

Shouldn't be prepended?

irregulator replied:
I think you're right; but somehow it worked here: https://ci.internal.exoscale.ch/blue/rest/organizations/jenkins/pipelines/puppet-omnibus/runs/6/nodes/17/steps/18/log/?start=0

exo-cedric replied:
Agree with @xgattuso ; if it works, it's confusing :wink:

irregulator replied:
Well, order doesn't play a role it seems.

```
➜  russell src/pkg/puppet-omnibus git:(fix_docker_registry) ✗ cat Makefile
.PHONY: my_target

my_target: $(info $(MYVAR))
        @:

➜  russell src/pkg/puppet-omnibus git:(fix_docker_registry) ✗ make MYVAR=la my_target 
la
➜  russell src/pkg/puppet-omnibus git:(fix_docker_registry) ✗ make my_target MYVAR=ko
ko
```

Will change it as you suggested, for the sake of clarity.

irregulator replied:
https://github.com/exoscale/puppet-omnibus/pull/2/commits/d864ed7a7e49bae00ce32dcb1d2b3cec07ebb324

exo-cedric commented on Makefile:
```diff
@@ -3,6 +3,7 @@ VERSION       := $(or ${puppet_version},"3.8.5")
 ITERATION     := $(or ${puppet_vendor_version},y7)
 PACKAGE_NAME  := "puppet-omnibus"
 PKG_ITERATION := exo5
+DOCKER_REGISTRY := $(or ${EXOSCALE_DOCKER_REGISTRY}, "registry.internal.exoscale.ch")
```
```suggestion
EXOSCALE_DOCKER_REGISTRY ?= "registry.internal.exoscale.ch"
```

exo-cedric commented on Makefile:
```diff
@@ -46,7 +47,7 @@ puppet_exo:
 docker: require_os
 	flock /tmp/puppet_omnibus_$(OS)_docker_build.lock \
-	docker build -f Dockerfile.$(OS) -t package_puppet_omnibus_$(OS) .
+	docker build -f Dockerfile.$(OS) -t package_puppet_omnibus_$(OS) --build-arg EXOSCALE_DOCKER_REGISTRY=$(DOCKER_REGISTRY) .
```
```suggestion
	docker build -f Dockerfile.$(OS) -t package_puppet_omnibus_$(OS) --build-arg EXOSCALE_DOCKER_REGISTRY=$(EXOSCALE_DOCKER_REGISTRY) .
```

irregulator committed:
Update Makefile

Co-authored-by: Cédric Dufour @ Exoscale <52404108+exo-cedric@users.noreply.github.com>

irregulator committed:
Update Makefile

Co-authored-by: Cédric Dufour @ Exoscale <52404108+exo-cedric@users.noreply.github.com>

Commit:
Jenkinsfile: input variable before target

This doesn't seem to actually change the make invocation; just
swap them for clarity.
